server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/mcp/mcpOauthClient.ts

```diff
@@ -353,17 +354,24 @@ export class OAuthClient {
             state: state,
         }).toString()
 
+        const urlStr = authz.toString()
+
         const opener =
             process.platform === 'win32'
                 ? {
                       cmd: 'cmd',
-                      args: ['/c', 'start', '', `"${authz.toString().replace(/"/g, '""')}"`],
+                      args: ['/c', 'start', '""', `"${urlStr.replace(/"/g, '""').replace(/%/g, '%%')}"`],
+                      opts: {
+                          detached: true,
+                          stdio: 'ignore',
+                          windowsVerbatimArguments: true,
+                      } as const,
                   }
                 : process.platform === 'darwin'
-                  ? { cmd: 'open', args: [authz.toString()] }
-                  : { cmd: 'xdg-open', args: [authz.toString()] }
+                  ? { cmd: 'open', args: [urlStr], opts: { detached: true, stdio: 'ignore' } as const }
+                  : { cmd: 'xdg-open', args: [urlStr], opts: { detached: true, stdio: 'ignore' } as const }
 
-        void spawn(opener.cmd, opener.args, { detached: true, stdio: 'ignore' }).unref()
+        spawn(opener.cmd, opener.args, opener.opts).unref()
```

Comment:
should probably be just the `window_showDocument` message since all the clients handle opening external browser correctly

also i do not think this logic as written will not work on VSC remote / JB remote